### PR TITLE
TermTreeSelector: Apply fix for term-selector hierarchy.

### PR DESCRIPTION
**Pull request by timmyc**
Pretty much a clone of the logic in #6617, so /ht to @aduth for the fix.  Much like the issue described there, given a heavily nested list of Terms, a case can arise in the `<TermTreeSelector />` where subsequent pages of terms are never properly requested.

__To Test__
To create the test scenario where this issue happens is a bit more difficult with terms as the default number returned in a page is 100.  As such, feel free to test with my test site `testingtimmy2` ( I think @nylen is a member of the site already ), by opening up a [new Project](http://calypso.localhost:3000/edit/jetpack-portfolio/testingtimmy2.wordpress.com) in the editor.

You can then watch the network tab to verify that two requests are made to the `terms` endpoint for the `jetpack-portfolio-type` taxonomy.  Much like #6617, it helps to see the breaking behavior first on `master` or `wpcalypso`, then test this branch out.

Test live: https://calypso.live/?branch=fix/term-tree-selector-hierarchy

### CODE REVIEW

**Comment by aduth** [comment]
How do you feel about using an approach like that in #6629 instead of the one from #6617 (which was closed in favor of the former)?

**Comment by timmyc** [comment]
This PR got bit by the meetup stagnation.  In 55e2bfd I applied the similar fix as what was done in #6629.

File: client/state/terms/utils.js [hunk 1]
```diff
@@ -30,3 +30,14 @@ export function getSerializedTermsQuery( query = {} ) {
...
 }
+
+/**
+ * Returns a serialized terms query, excluding any page parameter
+ *
+ * @param  {Object} query  Terms query
+ * @param  {Number} siteId Optional site ID
+ * @return {String}        Serialized terms query
+ */
+export function getSerializedTermsQueryWithoutPage( query, siteId ) {
```
Comment (aduth) [inline client/state/terms/utils.js]: You're not using `siteId`, I suspect copypasta error 😄

**Comment by aduth** [comment]
Minor item noted, but otherwise this looks good to go 👍

File: client/state/terms/utils.js (hunk 1 above)
Comment (timmyc) [inline client/state/terms/utils.js]: You are correct siiiir!

**Pull request merged**